homeassistant/components/template/number.py

```diff
@@ -47,16 +55,27 @@
             vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.template,
             vol.Required(CONF_STATE): cv.template,
             vol.Required(CONF_SET_VALUE): cv.SCRIPT_SCHEMA,
-            vol.Required(ATTR_STEP): cv.template,
-            vol.Optional(ATTR_MIN, default=DEFAULT_MIN_VALUE): cv.template,
-            vol.Optional(ATTR_MAX, default=DEFAULT_MAX_VALUE): cv.template,
+            vol.Required(CONF_STEP): cv.template,
+            vol.Optional(CONF_MIN, default=DEFAULT_MIN_VALUE): cv.template,
+            vol.Optional(CONF_MAX, default=DEFAULT_MAX_VALUE): cv.template,
             vol.Optional(CONF_OPTIMISTIC, default=DEFAULT_OPTIMISTIC): cv.boolean,
             vol.Optional(CONF_UNIQUE_ID): cv.string,
         }
     )
     .extend(TEMPLATE_ENTITY_AVAILABILITY_SCHEMA.schema)
     .extend(TEMPLATE_ENTITY_ICON_SCHEMA.schema)
 )
+NUMBER_CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): cv.template,
+        vol.Required(CONF_STATE): cv.template,
+        vol.Required(CONF_STEP): cv.template,
+        vol.Optional(CONF_SET_VALUE): cv.SCRIPT_SCHEMA,
```

Comment:
Why is set value optional? A number entity isn't allowed to not support the service to set the value.

Reply:
https://github.com/home-assistant/core/pull/124917